packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.page.tsx

```diff
@@ -140,50 +212,66 @@ const EditSectionPage = ({
 
 function questionTableRows(
   section: InferProps<typeof getServerSideProps>['section'],
-  applicationId: string
+  applicationId: InferProps<typeof getServerSideProps>['applicationId'],
+  handleOnUpDownButtonClick: () => void
 ) {
-  return (
-    section.questions?.map((question) => ({
-      cells: [
-        {
-          content: (
-            <p className="govuk-!-font-weight-bold">{question.fieldTitle}</p>
-          ),
-          className: 'govuk-!-padding-1',
-        },
-        {
-          content: <p>{ResponseTypeLabels[question.responseType]}</p>,
-          className: 'govuk-!-padding-1',
-        },
-        {
-          content: (
-            <div className="govuk-!-text-align-right govuk-width-container govuk-!-padding-top-2">
-              <div className="govuk-grid-row">
-                <div className="govuk-grid-column-one-quarter">
-                  <p />
-                </div>
-                <div className="govuk-grid-column-one-quarter">
-                  {/* GAP-2111: Uncomment when move question order is implemented */}
-                  {/* <Button text="Up" isSecondary /> */}
-                </div>
-                <div className="govuk-grid-column-one-quarter">
-                  {/* GAP-2111: Uncomment when move question order is implemented */}
-                  {/* <Button text="Down" isSecondary /> */}
-                </div>
-                <CustomLink
-                  href={`/build-application/${applicationId}/${section.sectionId}/${question.questionId}/edit/question-content`}
-                  customStyle="govuk-!-padding-left-6 govuk-!-padding-top-2 govuk-grid-column-one-quarter govuk-!-text-align-left"
-                >
-                  Edit
-                </CustomLink>
-              </div>
+  const questions = section.questions;
+  if (!questions) {
+    return [];
+  }
+
+  return questions.map((question, questionIndex) => ({
+    cells: [
+      {
+        content: (
+          <p className="govuk-!-font-weight-bold">{question.fieldTitle}</p>
+        ),
+        className: 'govuk-!-padding-1',
+      },
+      {
+        content: <p>{ResponseTypeLabels[question.responseType]}</p>,
+        className: 'govuk-!-padding-1',
+      },
+      {
+        content: (
+          <div className="govuk-!-text-align-right govuk-width-container govuk-!-padding-top-4 govuk-!-padding-right-2">
+            <div className="govuk-grid-row">
+              <button
+                className={`govuk-button govuk-button--secondary govuk-!-margin-right-2 govuk-!-margin-bottom-0 ${styles['button']}`}
+                data-module="govuk-button"
+                data-cy="cyUpButton"
+                name={`Up/${question.questionId}`}
+                disabled={questionIndex === 0}
+                onClick={handleOnUpDownButtonClick}
+                data-testid={`upButton-${questionIndex}`}
```

Comment:
Doesn't look like we're using these attributes?

Reply:
Which ones? Name, disabled, onClick & data-testid are all used

Reply:
Oh wait I see they're for the RTL stuff. I'm not a huge fan of having two different test ID attributes, could you not as easily do `getAllByRole` and use the array indexes to assert on which buttons are enabled? Eg:

```js
const upButtons = screen.getAllByRole('button', { name: 'Move section up' });
expect(upButtons[0]).toBeDisabled();
expect(upButtons[1]).not.toBeDisabled();
expect(upButtons[2]).not.toBeDisabled();

```

Reply:
Yeah sure - will give it a go